lib/logstash/outputs/elasticsearch.rb

```diff
@@ -293,6 +298,10 @@ def register
     # To support BWC, we check if DLQ exists in core (< 5.4). If it doesn't, we use nil to resort to previous behavior.
     @dlq_writer = dlq_enabled? ? execution_context.dlq_writer : nil
 
+    if dlq_enabled?
```

Comment:
if dlq is not enabled for a pipeline, and a user explicitly tells us how to route things to the DLQ, I would consider that a configuration error too.

Reply:
nice point!

Reply:
Fixed by 8e98c89a7d618698e7d2871001d39bec44857fa5